### DISCUSSION TITLE
readByQuery `sort` changed?

### DISCUSSION THREAD

**Discussion by chrisbartley** [directus/directus#11660] [Archive]
I recently upgraded from Directus 9.0.0-rc.93 to 9.5.1 and found that all my `readByQuery` calls which included a `sort` param were suddenly failing with the error `sortField.startsWith is not a function`. 🤔  For example, doing something similar to [the custom API endpoint recipe](https://docs.directus.io/extensions/endpoints/#example-recipes) now fails with that error:

```javascript
recipeService
   .readByQuery({ sort: [{ column: 'name', order: 'asc' }], fields: ['*'] })
   .then((results) => res.json(results))
   .catch((error) => {
      return next(new ServiceUnavailableException(error.message));
   });
```

I thought at first that changing `sort` to `sortField` fixed it.  But eventually realized that I might as well have changed it to `foobar` or just removed it altogether because although the error went away, so did the sorting :-\

I found no mention of this change anywhere in the release notes.  Am I correct that this is a recent change?  This most definitely used to work just fine.  How does one go about sorting results from `readByQuery` now?

Thanks!

**Comment by rijkvanzanten** [comment]
There have been a couple (breaking) tweaks and changes between the RC versions and the production release. The correct way to sort is through an array of strings, just like the API query parameters:

```
sort: ['name']
```

with field prefixed with `-` for descending search:

```
sort: ['-name']
```

**Reply by chrisbartley** [reply]
Docs fixed in PR #11662

**Reply by rijkvanzanten** [reply]
Thanks!

**Comment by mccount** [comment]
Guys sorry for responding to this, but I'm not sure why it's not working for me. Packages up to date (9.6.0).
![grafik](https://user-images.githubusercontent.com/7852259/157336920-1d549ea4-5e71-4e0e-9a69-b3ed9964f35b.png)
Could you help me?

**Reply by rijkvanzanten** [reply]
Looks like a typing issue in the SDK. Mind opening a new issue?

**Reply by stefanovalentegva** [reply]
I've got exactly the same error with "@directus/sdk": "^9.7.1"...

**Reply by philipp-winterle** [reply]
Same same 9.15.1

**Reply by azrikahar** [reply]
> Same same 9.15.1

@hummal Mind opening an issue in https://github.com/directus/sdk? Thanks!

**Reply by philipp-winterle** [reply]
https://github.com/directus/sdk/issues/40
